__tests__/app.test.js

```diff
@@ -181,4 +185,96 @@ describe('GET /api/topics', () => {
     })
+
+  })
+
+
+  describe('POST /api/articles/:article_id/comments', () => {
+    test('201: returns the posted comment', () => {
+      return request(app)
+        .post('/api/articles/1/comments')
+        .send({ username: 'butter_bridge', body: 'POST: comment/article_id' })
+        .expect(201)
+        .then(({ body }) => {
+          expect(body[0]).toEqual(
+            expect.objectContaining({
+              comment_id: 19,
+              body: 'POST: comment/article_id',
+              votes: 0,
+              author: expect.any(String),
+              article_id: 1,
+              created_at: expect.any(String),
+            })
+          );
+        });
+        })
+
+
+        test('201: ignores the unnecessary property', () => {
+          return request(app)
+            .post('/api/articles/1/comments')
+            .send({
+              username: 'butter_bridge',
+              body: 'POST: comment/article_id',
+              ignoreThis: 'not to be posted',
+            })
+            .expect(201)
+            .then(({ body }) => {
+              expect(body[0]).toEqual(
+                expect.objectContaining({
+                  comment_id: 19,
+                  body: 'POST: comment/article_id',
+                  votes: 0,
+                  author: expect.any(String),
+                  article_id: 1,
+                  created_at: expect.any(String),
+                })
+              );
+              expect(body[0]).not.toHaveProperty('ignoreThis');
+            });
+        })
+
+
+        test('400 Bad Request: invalid properties', () => {
+          return request(app)
+            .post('/api/articles/1/comments')
+            .send({random: 'butter_bridge', head: "POST: comment/article_id",})
+            .expect(400)
+            .then(({ body }) => {
+              expect(body.msg).toBe('Bad Request')
+            })
+        })
+
+        test('404 Not Found: valid but non-existing ID', () => {
+          return request(app)
+            .post('/api/articles/999/comments')
+            .send({ username: 'butter_bridge', body: 'POST: comment/article_id' })
+            .expect(404)
+            .then(({ body }) => {
+              expect(body.msg).toBe('Not Found');
+            })
+        })
+
+
+        test('400 Bad Request: invalid id (non-existing)', () => {
+          return request(app)
+            .post('/api/articles/not-an-ID/comments')
+            .send({ username: 'butter_bridge', body: 'POST: comment/article_id' })
+            .expect(400)
+            .then(({ body }) => {
+              expect(body.msg).toBe('Bad Request')
+            })
+        })
+
+
+        test('400 Bad Request: returns missing fields', () => {
+          return request(app)
+            .post('/api/articles/1/comments')
+            .send({ username: 'butter_bridge' })
+            .expect(400)
+            .then(({ body }) => {
+              expect(body.msg).toBe('Bad Request')
+            })
+        })
```

Comment:
Really nice and comprehensive test suite. There is one more test we could consider, how about if the username is not found in the database? (404)